packages/wrangler/src/triggers/deploy.ts

```diff
@@ -90,103 +93,110 @@ export default async function triggersDeploy(
 	const uploadMs = Date.now() - start;
 	const deployments: Promise<string[]>[] = [];
 
+	const deploymentInSync = deployToWorkersDev === available_on_subdomain;
+	const previewsInSync =
+		config.preview_urls === previews_available_on_subdomain;
+
 	if (deployToWorkersDev) {
 		// Deploy to a subdomain of `workers.dev`
 		const userSubdomain = await getWorkersDevSubdomain(
 			accountId,
 			config.configPath
 		);
-		const scriptURL =
+
+		const deploymentURL =
 			props.legacyEnv || !props.env
 				? `${scriptName}.${userSubdomain}.workers.dev`
 				: `${envName}.${scriptName}.${userSubdomain}.workers.dev`;
-		if (!available_on_subdomain) {
+
+		if (deploymentInSync && previewsInSync) {
+			deployments.push(Promise.resolve([deploymentURL]));
+		} else {
 			// Enable the `workers.dev` subdomain.
 			deployments.push(
 				fetchResult(`${workerUrl}/subdomain`, {
 					method: "POST",
-					body: JSON.stringify({ enabled: true }),
+					body: JSON.stringify({
+						enabled: true,
+						previews_enabled: config.preview_urls,
+					}),
 					headers: {
 						"Content-Type": "application/json",
 					},
-				})
-					.then(() => [scriptURL])
-					// Add a delay when the subdomain is first created.
-					// This is to prevent an issue where a negative cache-hit
-					// causes the subdomain to be unavailable for 30 seconds.
-					// This is a temporary measure until we fix this on the edge.
-					.then(async (url) => {
-						await sleep(3000);
-						return url;
-					})
```

Comment:
just to confirm, has this issue has been fixed?

Reply:
The subdomain is created automatically right away in the dashboard these days not just on first deploy. So yes, we do not see this issue anymore